handler/src/main/java/io/netty/handler/ssl/ReferenceCountedOpenSslEngine.java

```diff
@@ -0,0 +1,1951 @@
+        session.handshakeFinished();
+        engineMap.remove(ssl);
+        return FINISHED;
+    }
+
+    private SSLEngineResult.Status getEngineStatus() {
+        return engineClosed? CLOSED : OK;
+    }
+
+    private SSLEngineResult.HandshakeStatus mayFinishHandshake(SSLEngineResult.HandshakeStatus status)
+            throws SSLException {
+        if (status == NOT_HANDSHAKING && handshakeState != HandshakeState.FINISHED) {
+            // If the status was NOT_HANDSHAKING and we not finished the handshake we need to call
+            // SSL_do_handshake() again
+            return handshake();
+        }
+        return status;
+    }
+
+    @Override
+    public final synchronized SSLEngineResult.HandshakeStatus getHandshakeStatus() {
+        // Check if we are in the initial handshake phase or shutdown phase
+        return needPendingStatus() ? pendingStatus(SSL.pendingWrittenBytesInBIO(networkBIO)) : NOT_HANDSHAKING;
+    }
+
+    private SSLEngineResult.HandshakeStatus getHandshakeStatus(int pending) {
+        // Check if we are in the initial handshake phase or shutdown phase
+        return needPendingStatus() ? pendingStatus(pending) : NOT_HANDSHAKING;
+    }
+
+    private boolean needPendingStatus() {
+        return handshakeState != HandshakeState.NOT_STARTED && !isDestroyed()
+                && (handshakeState != HandshakeState.FINISHED || engineClosed);
+    }
+
+    /**
+     * Converts the specified OpenSSL cipher suite to the Java cipher suite.
+     */
+    private String toJavaCipherSuite(String openSslCipherSuite) {
+        if (openSslCipherSuite == null) {
+            return null;
+        }
+
+        String prefix = toJavaCipherSuitePrefix(SSL.getVersion(ssl));
+        return CipherSuiteConverter.toJava(openSslCipherSuite, prefix);
+    }
+
+    /**
+     * Converts the protocol version string returned by {@link SSL#getVersion(long)} to protocol family string.
+     */
+    private static String toJavaCipherSuitePrefix(String protocolVersion) {
+        final char c;
+        if (protocolVersion == null || protocolVersion.length() == 0) {
+            c = 0;
+        } else {
+            c = protocolVersion.charAt(0);
+        }
+
+        switch (c) {
+            case 'T':
+                return "TLS";
+            case 'S':
+                return "SSL";
+            default:
+                return "UNKNOWN";
+        }
+    }
+
+    @Override
+    public final void setUseClientMode(boolean clientMode) {
+        if (clientMode != this.clientMode) {
+            throw new UnsupportedOperationException();
+        }
+    }
+
+    @Override
+    public final boolean getUseClientMode() {
+        return clientMode;
+    }
+
+    @Override
+    public final void setNeedClientAuth(boolean b) {
+        setClientAuth(b ? ClientAuth.REQUIRE : ClientAuth.NONE);
+    }
+
+    @Override
+    public final boolean getNeedClientAuth() {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) A getX() method which returns a boolean should be named isX() [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://garage.netty.io/sonarqube/coding_rules#rule_key=pmd%3ABooleanGetMethodName)